src/mo_frontend/typing.ml

```diff
@@ -1975,15 +1975,17 @@ and infer_call env exp1 inst exp2 at t_expect_opt =
 
 (* Cases *)
 
-and infer_cases env t_pat t cases : T.typ =
-  List.fold_left (infer_case env t_pat) t cases
+and infer_cases env check_unused t_pat t cases : T.typ =
+  List.fold_left (infer_case env check_unused t_pat) t cases
 
-and infer_case env t_pat t case =
+and infer_case env check_unused t_pat t case =
```

Comment:
Do you need this extra `check_unused` parameter? I was about to suggest putting it as a field in env, but env.unused already exists. Can it not be re-used for this or is there something more subtle going on?

Reply:
I actually tried this yesterday as my first choice, but unfortunately some case failed. I believe it was in the base library, the `Trie.mo` where suddenly a false positive occurred (used declaration of a switch inside a declaration inside another switch was recognized as unused in error)... I will try to extract that case and add it to the test cases too. The reason is that the `env.unused` is applied recursively, while the `check_unused` is applied only at one level. And there is case where `env.unused` is recursively turned off, and then the local `check_unused` in a switch block should not override this. But it is really difficult...

Reply:
I try to see whether I can simplify this. I believe `check_unused` is actually `not env.pre` in this case. It is independent of `env.check_unused` as I checked yesterday...